From 26878890f66582fdcc5fa0076a56133c128a540c Mon Sep 17 00:00:00 2001
From: Yaroslav Halchenko <debian@onerussian.com>
Date: Thu, 7 Dec 2017 19:49:57 -0500
Subject: [PATCH] BF: resort to "ascii" handling logic if chardet guessing was
 incorrect

---
 src/dcmstack/extract.py | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/src/dcmstack/extract.py b/src/dcmstack/extract.py
index 48f8772..d34768f 100644
--- a/src/dcmstack/extract.py
+++ b/src/dcmstack/extract.py
@@ -292,12 +292,15 @@ def get_text(byte_str):
         if match['encoding'] is None:
             return None
         else:
-            return byte_str.decode(match['encoding'])
+            try:
+                return byte_str.decode(match['encoding'])
+            except UnicodeDecodeError:
+                pass
+
+    if not is_ascii(byte_str):
+        return None
     else:
-        if not is_ascii(byte_str):
-            return None
-        else:
-            return byte_str.decode('ascii')
+        return byte_str.decode('ascii')
 
 default_conversions = {'DS' : float,
                        'IS' : int,
-- 
2.15.0

